app/models/date_field_collection.rb

```diff
@@ -0,0 +1,33 @@
+# The goal of the class is to ensure that, if the user enters invalid values, like 35 January 2023,
+# we preserve what the user entered so we can show it back to them with an appropriate error message.
+# If we were to directly parse to a Date, invalid values would be rejected and at best we could
+# represent the entire attribute as `nil`, losing the specifics of what was invalid.
+class DateFieldCollection
+  def initialize(params, attribute_name)
```

Comment:
it's all right here,
I just wonder if we need the `attribute_name` since :dob is the only one.